model-optimizer/extensions/middle/GatherNdNormalizer.py

```diff
@@ -77,26 +78,24 @@ def replace_pattern(self, graph: Graph, match: dict):
         # 0. All needed checks that we can replace GatherNd by Gather
         gather_idx = self.indices_check(indices, input_shape)
         if gather_idx is None:
-            log.warning('Node {} with op=GatherNd  can\'t be normalized to op=Gather.'.format(gather.name))
+            log.warning('Node {} with op=GatherNd  can\'t be normalized to op=Gather.'.format(gather_name))
             return
 
         # 1. Add Reshape and connect
         new_shape = int64_array([-1] + list(input_shape[indices.shape[-1]:]))
-        reshape = Reshape(graph, {'name': gather.name + '/Reshape_for_GatherNd/'}).create_node()
-        reshape_const_node = Const(graph, {'name': reshape.name + '/Dim', 'value': new_shape}).create_node()
+        reshape = create_op_node_with_second_input(graph, Reshape, new_shape,
+                                                   {'name': gather_name + '/Reshape_for_GatherNd/'})
         gather.in_port(0).get_connection().set_destination(reshape.in_port(0))
-        reshape.in_port(1).connect(reshape_const_node.out_port(0))
 
         # 2. Change indices from Nd to 1d:
         new_indices = np.reshape(np.take(indices, indices=[gather_idx], axis=-1), [-1])
-        new_indices_const = Const(graph, dict(value=new_indices)).create_node()
-        axis_const = Const(graph, {'value': int64_array(0)}).create_node()
+
+        rename_node(gather, gather_name + '/to_delete')
 
         # 3. Create new Gather operation and reconnect all inputs/outputs
-        new_gather = Gather(graph, {'name': gather.name + '/NewGather/'}).create_node()
+        new_gather = create_op_with_const_inputs(graph, Gather, {1: new_indices, 2: int64_array(0)},
```

Comment:
You must call `rename_node` on `new_gather` to actually set a name `gather_name` to it.
Specifying a "name" for a node in a constructor when there is another node with id equal to that name is not enough.

Reply:
Done.